File: bookworm-backend/spring-boot-ecommerce/src/main/java/com/bookworm/ecommerce/dto/OrderDto.java
package com.bookworm.ecommerce.dto;

import com.bookworm.ecommerce.entity.OrderItem;
import lombok.Data;
import java.math.BigDecimal;
import java.util.Date;
import java.util.Set;

@Data
public class OrderDto {

    private Long id;
    private String orderTrackingNumber;
    private int totalQuantity;
    private BigDecimal totalPrice;
    private int status;
    private Date dateCreated;
    private Date lastUpdated;

    private Set<OrderItemDto> orderItems;
    private CustomerDto customer;
    private AddressDto billingAddress;
    private AddressDto shippingAddress;

}

File: bookworm_front/src/app/core/models/order-dto.ts
import { Customer } from './customer';
import { Address } from './address';
import { OrderItemDto } from './order-item-dto';

export class OrderDto {
  id: number;
  orderTrackingNumber: string;
  totalQuantity: number;
  totalPrice: number;
  status: any;
  dateCreated: Date;
  lastUpdated: Date;
  orderItems: OrderItemDto[];
  customer: Customer;
  billingAddress: Address;
  shippingAddress: Address;
}
